docs/modules/api/pages/index.adoc

```diff
@@ -302,22 +305,119 @@ Unlike `CheckResourceSet` -- which checks access to resource instances of the _s
   "action": "approve",
   "resourceKind": "leave_request",
   "policyVersion": "dev",
-  "filter": { <1>
-    "expression":  {
-      "operator":  "eq",
-      "operands":  [
-        { "variable":  "request.resource.attr.status" },
-        { "value":  "PENDING_APPROVAL" }
-      ]
+  "filter": {
+    "kind": "KIND_CONDITIONAL", <1>
+    "condition": { <2>
+        "expression":  {
+          "operator":  "eq",
+          "operands":  [
+            { "variable":  "request.resource.attr.status" },
+            { "value":  "PENDING_APPROVAL" }
+          ]
+        }
     }
   },
   "meta": { 
-    "filterDebug": "(request.resource.attr.status == \"PENDING_APPROVAL\")" <2>
+    "filterDebug": "(request.resource.attr.status == \"PENDING_APPROVAL\")" <3>
   },
 }
 ----
-<1> Abstract syntax tree (AST) of the policies' conditions, which have been partially evaluated using values provided in the request.
-<2> Filter AST represented as a string. Useful for debugging.
+<1> Filter kind can be `KIND_ALWAYS_ALLOWED`, `KIND_ALWAYS_DENIED` or `KIND_CONDITIONAL`. See below for description of what these values mean.
+<2> Populated only if `kind ` is `KIND_CONDITIONAL`. Contains the abstract syntax tree (AST) of the condition that must be satisfied to allow the action.
+<3> Condition AST represented as a human readable string. Useful for debugging.
+
+==== Structure of the `filter` block
+
+The `kind` field defines the filter kind.
+
+`KIND_ALWAYS_ALLOWED`:: The principal is unconditionally allowed to perform the action
+`KIND_ALWAYS_DENIED`:: The principal is unconditionally not permitted to perfrom the action
+`KIND_CONDITIONAL`:: The principal is allowed to perform the action if the condition is satisfied
+
+
+The `condition` field holds the AST of the condition that must be satisfied. It is rooted in an expression that has an `operator` (e.g. equals, greater than) and `operands` (e.g. a constant value, a variable or another expression).
+
+.Common Operators
+[caption=]
+[%header,cols=".^1m,.^4",grid=rows]
+|===
+| Operator | Description 
+| add               | Addition (+)
```

Comment:
Can we add `sub`, which indicate subtraction to this list? There's no example for it, but it is a very basic operator.

Reply:
There's a unit test for `sub` in `ast_test.go`.

Reply:
Ah! Good catch.